File: 2023/internal/day_02/day_02_test.go
package day_02

import (
	_ "embed"

	"testing"

	"github.com/stretchr/testify/require"
)

//go:embed sample_input.txt
var sample_input string


func TestPart1(t *testing.T) {
	require := require.New(t)

	// test on sample input
	require.Equal(8, Part1(sample_input))

	// test on real input
	require.Equal(2632, Part1(input))
}


func TestPart2(t *testing.T) {
	require := require.New(t)

	// test on sample input
	require.Equal(2286, Part2(sample_input))

	// test on real input
	require.Equal(69629, Part2(input))
}

File: 2024/src/day02/day02.test.ts
import { describe, it } from "@std/testing/bdd";
import { expect } from "@std/expect";
import { Day02Solution } from "./day02.ts";


describe("Day02Solution", () => {
	const solution = new Day02Solution();

	const sampleInput = [
		[7, 6, 4, 2, 1], // safe (decreasing)
		[1, 2, 7, 8, 9], // unsafe
		[9, 7, 6, 2, 1], // unsafe
		[1, 3, 2, 4, 5], // unsafe
		[8, 6, 4, 4, 1], // unsafe
		[1, 3, 6, 7, 9] // safe (increasing)
	];

	const realInput = solution.parseInput(Deno.readTextFileSync('src/day02/input.txt'));


	describe("parseInput()", () => {

		it("should return the correct value for the sample input", () => {
			const sampleInput = (
				"7 6 4 2 1\n" +
				"1 2 7 8 9\n" +
				"9 7 6 2 1\n" +
				"1 3 2 4 5\n" +
				"8 6 4 4 1\n" +
				"1 3 6 7 9\n"
			);

			expect(solution.parseInput(sampleInput)).toEqual([
				[7, 6, 4, 2, 1],
				[1, 2, 7, 8, 9],
				[9, 7, 6, 2, 1],
				[1, 3, 2, 4, 5],
				[8, 6, 4, 4, 1],
				[1, 3, 6, 7, 9]
			]);
		});

	});


	describe("part1()", () => {

		it("should return the correct value for the sample input", () => {
			expect(solution.part1(sampleInput)).toBe(2);
		});

		it("should return the correct value for the real input", () => {
			expect(solution.part1(realInput)).toBe(332);
		});

	});


});
